src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/LightweightReferenceToElementResolverTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.SearchValues;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Tests.Common;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Core.UnitTests.Features.Search
+{
+    public class LightweightReferenceToElementResolverTests
+    {
+        private readonly LightweightReferenceToElementResolver _resolver;
+
+        public LightweightReferenceToElementResolverTests()
+        {
+            ReferenceSearchValueParser referenceSearchValueParser = Mock.TypeWithArguments<ReferenceSearchValueParser>();
+
+            _resolver = new LightweightReferenceToElementResolver(referenceSearchValueParser, ModelInfoProvider.Instance);
+        }
+
+        [Fact]
+        public void GivenAValidReference_WhenConvertingAReferenceToTypedElement_ThenTheResultIsValid()
+        {
+            var patentRef = "Patient/1234";
+
+            var result = _resolver.Resolve(patentRef);
+
+            Assert.Equal("Patient", result.InstanceType);
+            Assert.Equal("1234", result.Children("id").Single().Value);
+        }
+
+        [Fact]
+        public void GivenAValidReferenceWithHistory_WhenConvertingAReferenceToTypedElement_ThenTheResultIsValid()
```

Comment:
Could this be made part of the above test by using InlineData params? It seems to be the same save for the patientRef value.